### PR TITLE
Use the navigate function in order to benefit from apollo cache

### PR DESCRIPTION
**Pull request by jeffersontuc**
#### Notes

This PR depends on and must be merged after the `checkout-resources` changes.

#### What is the purpose of this pull request?

As the title says, this PR adapts the minicart to use `render-runtime`'s navigate function.

#### What problem is this solving?

The checkout's cart now can benefit from the apollo cache.

#### How should this be manually tested?

- Open this [workspace](https://jeffaddepum--checkoutio.myvtex.com/)
- Add an item to cart
- Click on the cart icon at top right
- Proceed to cart
- Verify that the loading skeleton didn't appear

Alternatively, you can

- Add an item to cart
- Click on the cart icon at top right
- Open console dev tools
- Proceed to cart
- Verify that a request to get the `orderForm` is not being made

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Review by omninando — APPROVED** [review]

File: react/BaseContent.tsx
Comment (marcoskwkm) [inline react/BaseContent.tsx]: Navigation seems to be broken if the new checkout is not installed. I believe you should not use `navigate` in this case.

**Review by marcoskwkm — CHANGES_REQUESTED** [review]
Add the case when the button redirects to the current Checkout to your test plan.

File: react/BaseContent.tsx
Comment (marcoskwkm) [inline react/BaseContent.tsx]: You should also take into account the case where `finishShoppingButtonLink` is truthy and the account has the new Cart installed. In this case, I believe you should not use `navigate` either. In short, you should only use `navigate` if it sends the user to the new Cart.

**Review by marcoskwkm — DISMISSED** [review]

File: react/BaseContent.tsx
Comment (marcoskwkm) [inline react/BaseContent.tsx]: nit: I would make it explicit that we want `major > 0` instead of just a truthy value.

**Review by marcoskwkm — COMMENTED** [review]
LGTM, but just to be extra safe I'll approve this after you merge the changes in `checkout-resources`.

**Review by victorhmp — APPROVED** [review]

**Review by omninando — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]
Awesome! 👏

**Pull request merged**
